samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -904,6 +903,14 @@ class SamzaContainer(
     })
   }
 
+  def startDiagnostics {
+    if (containerContext.config.getDiagnosticsEnabled) {
+      info("Starting diagnostics.")
+      val diagnosticsAppender = Class.forName(containerContext.config.getDiagnosticsAppenderClass.get).
```

Comment:
It would be nice to capture the class loading exceptions and log a line here, then throw them as ConfigException(msg, e).